### PR TITLE
New component: Pane

### PR DESCRIPTION
**Pull request by mattlind**
Taking a shot at a new Pane control. This is being reviewed by the design team so there may be some changes for accessibility and responsive design still.

### CODE REVIEW

File: src/components/Pane/Pane.Props.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     * Mode of the pane.
+     * @default PaneMode.push
+     */
+    paneMode?: PaneMode;
+
+    /**
+     * Has the close button visible.
+     * @default true
+     */
+    closeButtonHidden?: boolean;
```
Comment (dzearing) [inline src/components/Pane/Pane.Props.ts]: Maybe the default here should be false. We've updated the naming guidelines in BESTPRACTICES.md to name the props the opposite of the default value. That way the user can predict the value. Also is consistent with how html attributes are named.

`closeButtonVisible`

File: src/demo/pages/PanePage/examples/Pane.Basic.Example.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                        Pane content goes here.
+                    </PaneContent>
+                </Pane>
+            </div>
+        );
+    }
+
+    @autobind
+    private _showPane() {
+        this.setState((prevState: IPaneSmallRightExampleState) => {
```
Comment (mattlind) [inline src/demo/pages/PanePage/examples/Pane.Basic.Example.tsx]: I think there is something I'm not understanding in VS Code.  Some files it auto-formats to 2 space and some to 4 space.  I see the workspace settings are set to 2 space but I haven't figured out why that isn't being followed.  Auto format on save is also turned on so this stuff just happens without warning.

File: src/components/Pane/Pane.Props.ts (hunk 1 above)
Comment (mattlind) [inline src/components/Pane/Pane.Props.ts]: Thanks, I like the naming guidelines and showed them to my own team as well.  I welcome the improvements.

**Comment by dzearing** [comment]
I tried using this last night in an example codepen (by pointing to my localhost bundle), but couldn't get it rendering like this:

```ts
<Fabric>
  <Pane>
   I'm stuff
  </Pane>
</Fabric>
```

**Comment by mattlind** [comment]
It expects the two child components, which you can see in my example page.  Panes should have two children, at least with previous pane controls I've used or built.  You're more than welcome to tell me a better design though.


--

Matt



From: David Zearing

Sent: Thursday, November 17, 12:39 PM

Subject: Re: [OfficeDev/office-ui-fabric-react] Pane control (#544)

To: OfficeDev/office-ui-fabric-react

Cc: Matt Lind, Author



I tried using this last night in an example codepen (by pointing to my localhost bundle), but couldn't get it rendering like this:


<Fabric> <Pane> I'm stuff </Pane> </Fabric>

**Comment by mattlind** [comment]
It expects the two child components, which you can see in my example page.  Panes should have two children, at least with previous pane controls I've used or built.  You're more than welcome to tell me a better design though.

```
<Pane>
    <WrappedContent>
        [ Content ]
    </WrappedContent>
    <PaneContent>
        [ Pane content ]
    </PaneContent>
</Pane>
```

**Comment by dzearing** [comment]
Friendly ping:
@battletoilet @Jahnp Can you have someone try/review this and give feedback?

**Comment by micahgodbolt** [comment]
@mattlind is this still a valid PR or should we close it up?

**Pull request closed without merging**